### DISCUSSION TITLE
Can MOOSE support silp boundary condition like as no-penetration boundary condition

### DISCUSSION THREAD

**Discussion by suqingji** [idaholab/moose#21418] [Q&A Modules: Navier-Stokes] [answered]
Dear, 
I want to simulate some non-sticky problems. So I need to set no-penetration boundary condition. What I'm wondering is that can MOOSE support silp boundary condition like as no-penetration boundary condition.
Looking forward to your reply, thank you.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Please have a look at the contact module documentation, especially the frictionless models
https://mooseframework.inl.gov/modules/contact/

Guillaume

**Reply by suqingji** [reply]
Thanks for your reply. I think I may not have described my problem clearly.
In fact, I want to simulate  non-sticky flow through `navier-stokes` module. So I need to set no-penetration boundary condition as shown below.
![image](https://user-images.githubusercontent.com/61961455/175771807-313f62e1-c0e3-4a84-848b-506e15ddc811.png)
Then I want to add wall function BC and no-penetration BC to k-ω or k-ε model. What I'm wondering is that can I set a no-penetration as a NodalBC or  IntegratedBC?
Looking forward to your reply, thank you.

**Reply by GiudGiud** [reply]
Hello

No flux is the default boundary condition, so ~~no-slip~~ no-penetration is the default.
So for FE you could leave it as is.

Guillaume

**Reply by suqingji** [reply]
Dear Prof.
Thank you for your reply.I think what I described may have led you to misunderstand.

In other words.I want to add a no penetration bc for momentum equation rather than a no flux or no slip bc.
As you mentioned above, no slip is /vec u=0, and no flux is /nabla u /cdot /vec n=/nabla v  /cdot /vec n = /nabla w  /cdot /vec n=0.
Here I write no penetration bc is /vec u /cdot /vec n = 0. These are different, so I want to consult you if moose have no penetration bc.

Looking forward toyour reply. Thank you for your reply and help again.

**Reply by suqingji** [reply]
Dear Prof.
Please allow me to add a little description as shown below.

![image](https://user-images.githubusercontent.com/61961455/175798929-611818a8-14d7-4ed4-9a47-988ee577c4fb.png)

**Reply by GiudGiud** [reply]
the last one is no diffusive flux. 
no-penetration is the same as no advective flux

If you want you can add a boundary condition that does nothing. Simply not executing advection kernels on the boundary nodes is enough to have 0 advection there

**Reply by suqingji** [reply]
Dear Prof.
Thank you for your reply. 
If I understand you correctly, you mean to introduce the terms of advective flux by integration by parts to advective term. Here's how I did it.
![image](https://user-images.githubusercontent.com/61961455/175854864-2437e6c8-bbfd-4a24-9358-313c593903fd.png)

If this method is used, the outlet must be given a advection boundary condition. Unfortunately, I found this method to be less robust than the current method. And I think that's probably why by default MOOSE don't integrate by parts for advection terms.

Therefore I'm wondering whether there are other robust methods to achieve no penetration BC on MOOSE.

Looking forward toyour reply. Thank you for your reply and help again.

**Reply by GiudGiud** [reply]
In INS-FE both approaches have been implemented. I dont know that a conclusion was made as to which approach is more robust. But we moved to finite volume for most of the Navier Stokes work, where we do integrate that term over faces (eg basically an integration by part)

**Reply by suqingji** [reply]
Thank you,

This is worth studying. As far as I know, the COMSOL turbulence module has these functions.
